toolkit/tools/pkg/imagecustomizerlib/customizeoverlay.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+func enableOverlay(overlays *[]imagecustomizerapi.Overlay, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if overlays == nil {
+		return nil
+	}
+
+	// Integrate overlay dracut module and overlay driver into initramfs img.
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	overlayDracutModule := "overlay"
+	overlayDracutDriver := "overlay"
+	err = buildDracutModule(overlayDracutModule, overlayDracutDriver, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Dereference the pointer to get the slice
+	overlaysDereference := *overlays
+	err = updateGrubConfigForOverlay(imageChroot, overlaysDereference)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateGrubConfigForOverlay(imageChroot *safechroot.Chroot, overlays []imagecustomizerapi.Overlay) error {
+	var err error
+	var overlayConfigs []string
+
+	// Iterate over each Overlay configuration
+	for _, overlay := range overlays {
+		formattedPartition, err := systemdFormatPartitionId(overlay.Partition.IdType, overlay.Partition.Id)
```

Comment:
Is the `Partition` property optional or required?

Looking at this code, it appears to be required. But the config has the type as a pointer (`*VerityPartition`) and your doc has an example that doesn't provide a value for `Partition`.

Reply:
Added additional checking for Partition field.